bpf/lib/nodeport.h

```diff
@@ -873,6 +894,11 @@ int tail_nodeport_dsr_ingress_ipv6(struct __ctx_buff *ctx)
 	case CT_NEW:
 	case CT_REOPENED:
 create_ct:
+		if (port == 0) {
+			ret = DROP_INVALID;
+			goto drop_err;
+		}
```

Comment:
Couldn't we check for this case and drop the packet sooner, when we read the port before the tail call?

Reply:
See the comment in `tail_nodeport_dsr_ingress_ipv4()` - this isn't about the TCP-SYN containing bad DSR info, but about a subsequent TCP packet that expects a matching CT entry but doesn't find it (even though we checked earlier that it exists). It's arguably the most hacky/tricky part of the PR :/.

Reply:
OK. I think I get it.